app/services/resource_service.rb

```diff
@@ -15,101 +15,159 @@ def create(resource_class, params)
     # das Attribute 'visible' wird in den params mitgegeben und ist 'true'
-    if data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
+    if @data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
       @params = { visible: false }.merge(@params)
     end
 
     @resource_class = resource_class
     @resource = resource_class.new(@params)
-    @resource.data_provider = data_provider
+    @resource.data_provider = @data_provider
 
     # skip create if record already exists and new record has the same attribute values as the new
     # record and the record is not marked as 'always_recreate'
-    external_resource = find_external_resource
-    old_record = external_resource.try(:external)
-    if external_resource.present? && unchanged_attributes?(old_record) && !always_recreate?(data_provider, resource_class)
-      external_resource.touch
-      return old_record
+    @external_resource = find_external_resource
+    @old_resource = find_old_resource
+    if @external_resource.present? && unchanged_attributes? && !always_recreate?
+      @external_resource.touch
+      return @old_resource
     end
 
-    create_or_recreate(old_record, external_resource)
-    resource_or_error_message(resource)
+    resource_or_error_message(update_or_create)
   end
 
-  def create_or_recreate(old_record, external_resource)
-    ActiveRecord::Base.transaction do
-      old_record.destroy if old_record.present?
-      external_resource.destroy if external_resource.present?
-      if @resource.save
-        create_external_resource
-        FacebookService.delay.send_post_to_page(resource_id: @resource.id, resource_type: @resource_class.name)
-        set_default_categories if @resource.respond_to?(:categories)
+  private
+
+    # Find old resource
+    #
+    # @return [Object] Instance of @resource_class
+    def find_old_resource
+      if @old_resource_id.present?
+        @resource_class.filtered_for_current_user(@data_provider.user).find_by(id: @old_resource_id)
       else
-        GraphQL::ExecutionError.new("Invalid input: #{@resource.errors.full_messages.join(', ')}")
+        @external_resource.try(:external)
       end
     end
-  end
 
-  def unchanged_attributes?(record)
-    return false if record.blank?
-    return @resource.compareable_attributes == record.compareable_attributes if record.respond_to?(:compareable_attributes)
+    def update_or_create
+      if @old_resource.present?
+        update_resource
+      else
+        create_new_resource
+      end
+    end
 
-    # Fallback, wenn :compareable_attributes nicht beim Model definiert ist.
-    except_attributes = ["id", "created_at", "updated_at", "tag_list", "category_id", "region_id", "visible"]
-    @resource.attributes.except(*except_attributes) == record.attributes.except(*except_attributes)
-  end
+    def update_resource
+      # find all association names to delete for a resource
```

Comment:
Wir müssen immer alle Params mitschicken, damit die nested neu angelegt werden. Fehlende Params, werden als "zu löschen" verstanden.

Reply:
Kommentar ergänzt in 0215c59